gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -476,35 +484,70 @@ func (disablerTransform *commandDisabler) observeAndWriteCommand(cmd api.Cmd) er
 	return disablerTransform.writeCommand(cmd)
 }
 
-func (disablerTransform *commandDisabler) doesContainDisabledCmd(id api.SubCmdIdx) bool {
-	for _, r := range disablerTransform.disabledCommands {
-		if id.Contains(r) {
-			return true
+func (disablerTransform *commandDisabler) sortAndMergeDisabledCmds(ctx context.Context) {
+	/*
+		This functions ensures these conditions:
+			- All disabled commands are sorted so we can guarantee that any command to be
+			disabled is not touched by command disabler, therefore can be assumed to have
+			correct id.
+
+			- Ensures for this input set:
+			"[3893.0.6.3.0.13], [3893.0.6.3.0.17], [3893.0.6.3.0.22], [3893.0.6.3]"
+			All the subcommands of [3893.0.6.3] are eliminated as they are not necessary.
+
+			- Removes all duplicates.
```

Comment:
typo: This function (no "s") ensures ...

Suggested rephrasing:
```
- To-be-disabled commands are first sorted and duplicates are removed
  such that the actual command disabling can happen in a single sweep
  of the original command buffer, with the relevant command IDs.

- In the list of commands to be disabled, commands that are subcommands of others are removed.
  E.g., "[1.0.2.3], [1.0.2.4], [1.0.2]" is cleaned up to "[1.0.2]", since disabling this command also disables
  its subcommands.
```